packages/grafana-ui/src/components/GraphNG/types.ts

```diff
@@ -0,0 +1,13 @@
+import { DataFrame, Field } from '@grafana/data';
+
+export enum GraphNGLegendEventMode {
+  select = 'select',
+  append = 'append',
+}
+
+export interface GraphNGLegendEvent {
+  field: Field;
+  frame: DataFrame;
+  data: DataFrame[];
```

Comment:
As we discussed I think we could use indexes instead of passing the entire shebang here:)